services/products_links/get_products_links.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+class GetProductsLinks
+  RETRY_INTERVAL = 0.01
+  TIMEOUT = 5
+
+  class NotFoundError < StandardError; end
+
+  def initialize(url:, tag:)
+    @url = url
+    @tag = tag
+    @browser = Ferrum::Browser.new
+  end
+
+  def parse_links_without_js
+    doc = Nokogiri::HTML(URI.open(@url))
```

Comment:
why we should use `Nokogiri` here? Is it possible to use `Ferrum` everywhere?

Reply:
Nokogiri is more efficient and faster. It just goes to the page and gives the html, while Ferrum brings up a whole invisible browser window to read from it. But, unfortunately, Nokogiri doesn't know how to work with JS, that why we need Ferrum